File: OC/WGAnimation/WGAnimation/main/ViewController.m
//
//  ViewController.m
//  WGAnimation
//
//  Created by wanggang on 2018/8/17.
//  Copyright © 2018年 wanggang. All rights reserved.
//


#import "ViewController.h"
#import "WGClockViewController.h"
#import "WGLearnConceptViewController.h"
#import "WGTranslationViewController.h"
#import "WGSmileFaceViewController.h"
#import "WGHeartViewController.h"
#import "WGCAKeyframeAnimationViewController.h"
#import "WGMaskViewController.h"
#import "WGMenuViewController.h"
#import "WGBezierPathViewController.h"

@interface ViewController ()<UITableViewDelegate, UITableViewDataSource>

@property (nonatomic, strong) UITableView *tableView;
@property (nonatomic, strong) NSArray *dataArr;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.title = @"ViewController";
    [self buildView];
    [self loadData];
}

- (void)buildView{
    [self.view addSubview:self.tableView];
}

- (void)loadData{
    self.dataArr = @[@"实验一些概念", @"绘制时钟", @"创建平移/缩放/旋转动画", @"笑脸", @"心跳", @"关键帧动画", @"mask遮罩", @"扇形菜单", @"贝塞尔曲线"];
}

#pragma mark -<UITableViewDelegate, UITableViewDataSource>
-(NSInteger)numberOfSectionsInTableView:(UITableView *)tableView{
    return 1;
}

-(NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section{
    return self.dataArr.count;
}

-(UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath{
    
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:NSStringFromClass([UITableViewCell class]) forIndexPath:indexPath];
    
    cell.textLabel.text = [self.dataArr objectAtIndex:indexPath.row];
    return cell;
}

-(void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath{
    [tableView deselectRowAtIndexPath:indexPath animated:YES];
    
    switch (indexPath.row) {
        case 0:{
            WGLearnConceptViewController *vc = [[WGLearnConceptViewController alloc] init];
            [self.navigationController pushViewController:vc animated:YES];
            break;
        }case 1:{
            WGClockViewController *vc = [[WGClockViewController alloc] init];
            [self.navigationController pushViewController:vc animated:YES];
            break;
        }case 2:{
            WGTranslationViewController *vc = [[WGTranslationViewController alloc] init];
            [self.navigationController pushViewController:vc animated:YES];
            break;
        }case 3:{
            WGSmileFaceViewController *vc = [[WGSmileFaceViewController alloc] init];
            [self.navigationController pushViewController:vc animated:YES];
            break;
        }case 4:{
            WGHeartViewController *vc = [[WGHeartViewController alloc] init];
            [self.navigationController pushViewController:vc animated:YES];
            break;
        }case 5:{
            WGCAKeyframeAnimationViewController *vc = [[WGCAKeyframeAnimationViewController alloc] init];
            [self.navigationController pushViewController:vc animated:YES];
            break;
        }case 6:{
            WGMaskViewController *vc = [[WGMaskViewController alloc] init];
            [self.navigationController pushViewController:vc animated:YES];
            break;
        }case 7:{
            WGMenuViewController *vc = [[WGMenuViewController alloc] init];
            [self.navigationController pushViewController:vc animated:YES];
            break;
        }case 8:{
            WGBezierPathViewController *vc = [[WGBezierPathViewController alloc] init];
            [self.navigationController pushViewController:vc animated:YES];
            break;
        }
        default:
            break;
    }
}

#pragma mark -懒加载
-(UITableView *)tableView{
    if (!_tableView) {
        _tableView = [[UITableView alloc] initWithFrame:CGRectMake(0, 0, WGWIDTH, WGHEIGHT) style:UITableViewStylePlain];
        _tableView.backgroundColor = [UIColor whiteColor];
        _tableView.delegate = self;
        _tableView.dataSource = self;
        [_tableView registerClass:[UITableViewCell class] forCellReuseIdentifier:NSStringFromClass([UITableViewCell class])];
    }
    return _tableView;
}

-(NSArray *)dataArr{
    if (!_dataArr) {
        _dataArr = [[NSArray alloc] init];
    }
    return _dataArr;
}











@end

File: Swift/内容层动画/WGCoreAnimation/WGCoreAnimation/main/ViewController.swift
//
//  ViewController.swift
//  WGCoreAnimation
//
//  Created by wanggang on 2019/4/23.
//  Copyright © 2019 bozhong. All rights reserved.
//

import UIKit

class ViewController: UIViewController {
    
    let tableView = UITableView.init(frame: CGRect.init(x: 0, y: 0, width: WgWith, height: WgHeight))
    let dataArr = ["CALayer层动画", "水纹动画", "渐进圆环动画", "大火动画", "霓虹动画", "大雪动画", "扫描动画", "跳动动画"]
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        title = "列表"
        
        tableView.register(UITableViewCell.self, forCellReuseIdentifier: "UITableViewCell")
        tableView.delegate = self
        tableView.dataSource = self
        view.addSubview(tableView)
    }
    
}

//MARK:- UIViewTableView DataSource & Delegate
// 知识点：扩展
// 扩展和 Objective-C 中的分类类似，但没有名称
// 扩展可以为一个类型添加新的功能，但是不能重写已有的功能。
// 扩展可以添加新的计算型属性，但是不可以添加存储型属性，也不可以为已有属性添加属性观察器。
extension ViewController: UITableViewDelegate, UITableViewDataSource{
    
    func numberOfSections(in tableView: UITableView) -> Int {
        return 1
    }
    
    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return dataArr.count;
    }
    
    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        
        let cell = tableView.dequeueReusableCell(withIdentifier: "UITableViewCell", for: indexPath)
        cell.textLabel?.text = dataArr[indexPath.row]
        return cell
    }
    
    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        tableView.deselectRow(at: indexPath, animated: true)
        
        switch indexPath.row {
        case 0:
            let vc = WGCALayerAnimationViewController()
            self.navigationController?.pushViewController(vc, animated: true)
            
        case 1:
            let vc = WGWaterWaveBtnViewController()
            self.navigationController?.pushViewController(vc, animated: true)
            
        case 2:
            let vc = WGCircleBtnAniViewController()
            self.navigationController?.pushViewController(vc, animated: true)
            
        case 3:
            let vc = WGFireAniViewController()
            self.navigationController?.pushViewController(vc, animated: true)
            
        case 4:
            let vc = WGNeonAniViewController()
            self.navigationController?.pushViewController(vc, animated: true)
            
        case 5:
            let vc = WGSnowAniViewController()
            self.navigationController?.pushViewController(vc, animated: true)
            
        case 6:
            let vc = WGScanAniViewController()
            self.navigationController?.pushViewController(vc, animated: true)
            
        case 7:
            let vc = WGBeatAniViewController()
            self.navigationController?.pushViewController(vc, animated: true)
            
        default:
            break
        }
    }
}
